### ISSUE TITLE
Create a VGA connector based extension board for the Atlys Board

### ISSUE THREAD

**Issue by mithro** [timvideos/HDMI2USB-jahanzeb-firmware#1]
VGA is going to be around for a long time, we really need to continue to have a VGA input system.

**Comment by mithro** [comment]
Work was started on this project at https://github.com/timvideos/HDMI2USB-vmodvga but it is currently in limbo.

This board was based around [AD9984A](http://www.analog.com/en/audiovideo-products/analoghdmidvi-interfaces/ad9984a/products/product.html) device.

> The AD9984A is a complete 10-bit, 170 MSPS, monolithic analog interface optimized for capturing YPbPr video and RGB graphics signals. Its 170 MSPS encode rate capability and full power analog bandwidth of 300 MHz support all HDTV video modes up to 1080p, as well as graphics resolutions up to UXGA (1600 × 1200 at 60 Hz).
> 
> The AD9984A includes a 170 MHz triple ADC with an internal reference, a PLL, and programmable gain, offset, and clamp control. The user provides only a 1.8 V power supply and an analog input. Three-state CMOS outputs can be powered from 1.8 V to 3.3 V.

Rough pricing puts it at around $15 USD for individual quantities. http://findchips.com/search/ad9984a

**Comment by mithro** [comment]
Comment by @rohit91

> For, VGA Capturing, we need a fast ADC(3 channels) or specially made ADC ICs for VGA
> capturing like TVP700(http://www.ti.com/product/tvp7002). There is a nice & properly
> documented project here which does exactly that using a CPLD. Not a tough task to
> implement it on HDMI2USB: http://www.rpg.fi/desaster/blog/2013/04/19/vga-framegrabbing-with-tvp7002/ 

Rough pricing puts the TVP7002 at around $7 USD for individual quantites. http://www.findchips.com/search/tvp7002

**Comment by mithro** [comment]
Take a look at https://github.com/desaster/grabor

**Comment by mithro** [comment]
The VGA capture board created by @rohit91 can be found at https://github.com/timvideos/HDMI2USB-vmodvga

@rohit91 Can you link to your final status report here.

What do you think is left before we can close this issue?

**Comment by rohitk-singh** [comment]
Project Status Report: http://dreamsxtrinsic.blogspot.com/2014/08/gsoc-project-report-vga-capture.html

Since, the issue is to "Create VGA connector based extension board" hardware which is done. We can close this issue now, and create another issue for the required (specific) software issues for the board. If need arises in future, we can re-open this

**Comment by mithro** [comment]
Repository can be found at https://github.com/timvideos/HDMI2USB-vmodvga

From @rohit91 's blog;

<blockquote>
<h2>Original Goals</h2>
<p>
The goal of this project was to develop a VGA Capture expansion board for HDMI2USB on Digilent Atlys. The deliverables for the project were usable product with associated hardware, software and documentation.
</p><p>
Fine details of original goals were:
<ul>
 <li>Working PCB for VGA Capture
 <li>Working VGA Capture code for the hardware
 <li>Integration of this code with HDMI2USB
 <li>EDID Capability for the VGA Capture expansion board
</ul>
<p>
Original Proposal: <a href="http://goo.gl/rRtSrS">http://goo.gl/rRtSrS</a>
</p>
<h2>Final Outcome</h2>
<p>
I achieved quite significant success, along with associated failures, in this project. While I failed to reach all the original proposed goal I have succeeded in getting the following working:
<ul>
 <li>Nice and working PCB for VGA Capturing using Atlys
 <li>Working, but not perfect, VGA Capture HDL code
 <li>Integrated the VGA capture code into HDMI2USB.
</ul>
</p><p>
The Expansion PCB features a VGA digitizing IC from Analog Devices, AD9984A and has support for auto-detection and as well as EDID. Its design is production-ready unless issues crop in future during continued development on this project.
</p><p>
The VGA capture code, on other hand, needs improvement and ruggedness to field working environment. It is currently working very well with the latest PCB version, but continued development is required on it.
</p><p>
Video of current progress: https://www.youtube.com/watch?v=zSpOugDc2fM
</p>
</blockquote>

Images from the post;
![image](https://cloud.githubusercontent.com/assets/21212/5524514/45ed4b36-8a26-11e4-8a1f-1b770687b526.png)
![image](https://cloud.githubusercontent.com/assets/21212/5524518/5090a5c4-8a26-11e4-9a8c-d6b95a19d281.png)
![image](https://cloud.githubusercontent.com/assets/21212/5524522/5c7e751e-8a26-11e4-9efa-4edb9cdfeca8.png)

**Comment by mithro** [comment]
The [HDMI2USB-jahanzeb-firmware](https://github.com/timvideos/HDMI2USB-jahanzeb-firmware) is no longer under development. 

All active development is now on the [HDMI2USB-misoc-firmware](https://github.com/timvideos/HDMI2USB-misoc-firmware) which supports both the [Atlys](https://hdmi2usb.tv/digilent-atlys/) and [Opsis](https://hdmi2usb.tv/numato-opsis/) boards and has loads of other new features!
